File: 0501_0600/0578/0578.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n = int(f.read())
    s = []
    while n:
        k = n % 3 if n % 3 else 3
        s.insert(0, str(k))
        n = (n - k) // 3
    q.write(''.join(s))

File: 0501_0600/0578/0578.c
#include <stdio.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, k;
	char s[21], *t = s + 20;

	*t = '\0';
	fscanf(f, "%d", &n);
	while (n)
	{
		k = n % 3 ? n % 3 : 3;
		*--t = '0' + k;
		n = (n - k) / 3;
	}
	fprintf(q, t);
	return 0;
}
